docs/release_guidelines.md

```diff
@@ -1,23 +1,115 @@
+
+The second goal is greater consistency in the framework. Now that we have release branches that exist for testing older versions of the product, we don't need to carry logic forward to handle every version forever. We have many areas of the code that are performing checks on the product version and then performing actions if we are deploying or testing a version greater than a specific version. This will no longer be necessary once it can be assumed that the latest code will always include a feature and so the code that is performing this version check can be removed. This will enable us to reduce the amount of code in the framework that simply exists to make sure our framework is doing something that the product supports based on version, reducing a lot of technical debt.
+
+Overall, implementing release branches should allow us greater stability in testing existing releases as well as reducing the complexity of some of our framework code.
+
+## Supported versions and use-cases
+
+Release branches will support deployments and testing of ODF/OCP versions N and N-1, N being the version specified in the name of the branch.
+
+N: Standard use-case of deployment and testing the version specified in the release branch name.
+N-1: For upgrade purposes. We need to be able to deploy N-1, run pre-upgrade tests, upgrade and reload configs for version N, then run post-upgrade tests. This process requires that release branches support the deployment and testing of previous versions.
+
+For example, the branch release-4.8 will be able to deploy OCP/ODF 4.8 as well as 4.7 to support upgrade testing. However, it will not officially support deployment or testing of versions <= 4.6.
+
+
+## Usage
+
+### Locally
+
+As these are simply git branches we are creating, you just need to perform a git checkout on the specific release branch you wish to run your deployment or tests with in order to use it.
+
+```
+git fetch upstream
+git checkout release-4.12
+```
+
+Note that `upstream` is the name of the remote for the base repository, not your personal fork.
+
+### Jenkins
+
+From Jenkins we will automatically be using the release branch if no other ocs-ci branch has been specified. Which specific release branch we use will be dynamically based on the `OCS_VERSION` or `UPGRADE_OCS_VERSION` parameters. For example, if `OCS_VERSION` is 4.13 we will use `release-4.13`. If this release branch doesn't exist yet, we will default back to the `stable` branch. This logic can be overridden by specifying a branch to use.
+
+## Creating Release Branches
```

Comment:
would be good to say that branches are planned to be created around the GA time of the corresponding ODF release.
and also, that it's a one-off thing and there is no expectation that people would create branches regularly on their own

Reply:
I emphasized that the branch creation process is done by the project maintainers and done in conjunction with the ODF GA